common/src/main/java/com/wynntils/features/TerritoryDefenseMessageFeature.java

```diff
@@ -37,10 +46,32 @@ public void onInventoryClick(InventoryMouseClickedEvent event) {
             Matcher matcher = StyledText.fromComponent(tooltipLine)
                     .getMatcher(TERRITORY_DEFENSE_PATTERN, PartStyle.StyleType.NONE);
             if (matcher.matches()) {
-                Handlers.Command.sendCommandImmediately(
-                        "g %s defense is %s".formatted(titleMatcher.group(1), matcher.group(1)));
+                // intentionally not localized to match Wynncraft language
+                territoryMessages.put(
+                        titleMatcher.group(1),
+                        Pair.of(
+                                System.currentTimeMillis(),
+                                "g %s defense is %s".formatted(titleMatcher.group(1), matcher.group(1))));
                 return;
             }
         }
     }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onChatMessage(ChatMessageReceivedEvent e) {
```

Comment:
Sounds like this would also send the defence in this case: 
- A clicked attack, it failed, defense was saved
- B clicked attack, it happened, both send the defence.

So, perhaps, this needs to be a timed map. I would also advise against using a `Pair<Long, String>`. At least, make it a `Pair<Long, TerritoryDefense(idr the name)>`, or optionally, add a private record to this model that holds a `long timestamp, TerritoryDefense defense`. Lastly, this was my bad originally, but please extract the format string to a constant, while you are at it. Thanks.

Reply:
How would a timed map fix this? In both situations, if a second person attacks the same territory within MESSAGE_TIMEOUT, both will send it. We can adjust this to be lower, but it might not send when the server is laggy then. But if someone else attacks after MESSAGE_TIMEOUT, the first person won't send it.

Reply:
The sound solution does work. I've just finished it now